src/main/fc/rc.c

```diff
@@ -754,9 +758,11 @@ FAST_CODE_NOINLINE void updateRcCommands(void)
     for (int axis = 0; axis < 3; axis++) {
 
         float tmp = MIN(fabsf(rcData[axis] - rxConfig()->midrc), 500.0f);
+        // larger deadband on pitch and roll when in position hold
+        const float tmpDeadband = (FLIGHT_MODE(POS_HOLD_MODE) && rcControlsConfig()->pos_hold_deadband) ? rcControlsConfig()->pos_hold_deadband * 5.0f : rcControlsConfig()->deadband;
```

Comment:
rcControlsConfig()->pos_hold_deadband * 5.0f is a bit cryptic.
It's because the deadband is expressed in percent, and a 100% range spans, essentially, 500 rc units.
So `*5.0f* is shorthand for `rcControlsConfig()->pos_hold_deadband / 100.0f * 500.0f`.
If we had an init function for RC controls, it might make sense to put that maths in such a function.

I don't like this method at all.

A much better approach would be for the sticks to change the target location. That would be preferable for a number of reasons.  I just couldn't figure out how to do it.  This was the easiest method for me to implement at this point in time.

Reply:
> Is because the deadband is expressed in percent,

There is deadband expressed in PWM counts and (pos_hold)deadband in %.

> A much better approach would be for the sticks to change the target location. 

Can you describe how should it work (ideally as user documentation for 'position adjustment using sticks')?

Reply:
> There is deadband expressed in PWM counts and (pos_hold)deadband in %.
Ah, I understand now.  Yes this deadband could be expressed in PWM counts, but I thought percentage would be easier for the 'ordinary user; to understand, in this situation, where 10 means 10% of stick travel.  Otherwise they have to enter '50' to mean 10% deadband.

> how should it work (ideally as user documentation for 'position adjustment using sticks')?

I was thinking that, for example, if:
- the quad was facing due North, and if user pushed pitch forward, we add some amount to the current target position latitude value.
- the quad was pointing West, and user pitches pitch forward, we subtract some amount from the current target position longitude value.
- then we update our position target.

I imagine that we would need to compare the stick 'angle' (ie, pitch forward = 0 degrees, pitch back = 180 degrees), to the current heading of the quad, and from this we proportion the current stick deflection to the latitude and longitude for our next target position.

Functionally, the stick input would then command a change in position over time, ie a velocity.  And the PIDs would control the error, so the system becomes a velocity controller. Then, as the sticks are returned towards centre, the PIDs would smoothly slow the quad down, and we will should get a nice smooth stop.

It's possible that flight won't be so smooth like this, and that it might not work at all.  Since movement relies on somewhat laggy PIDs, it may be very vague.  The biggest issue may be situations where the quad is yawed quickly, in which case things may not be so smooth.

But it would probably allow me to remove that strange code that resets the target once the quad reverses direction, since the controller probably won't overshoot nearly as much it will start decelerating as soon as the user starts to return the stick towards the centre.

I am just curious to know what it might fly like.  Certainly it won't be 'dynamic', it will be sluggish, but it might be quite smooth and intuitive.  I just don't know. If the intent is to simply adjust the position of the quad in space, I think it would work better than Angle mode does.  And definitely be smoother.

Thanks for offering to help!